test/index.test.js

```diff
@@ -120,4 +120,42 @@ describe('aliases', () => {
     expect(await jarvis.send('hello "John Doe" how are you'))
       .toEqual('Hello John Doe');
   });
-});
\ No newline at end of file
+});
+
+describe("Macros", async () => {
+  const jarvis = new Jarvis();
+  jarvis.addCommand({
+    command: "greet <name>",
+    aliases: ["hello <name>"],
+    handler: ({ args }) => {
+      return `Hello ${args.name},how are you?`;
+    }
+  });
+
+  jarvis.addCommand({
+    command: "reply <name>",
+    handler: ({ args }) => {
+      return `Hello ${args.name},i'm fine thank you!`;
+    }
+  });
+
+  jarvis.addCommand({
+    command: "end <name>",
+    handler: ({ args }) => {
+      return `Good it was nice meeting you ${args.name}`;
+    }
+  });
+
+  jarvis.addMacro({
```

Comment:
actually the intended usage for macros is from the command line, not from code. e.g.

```
$ how to introduce $name
>> you are now entering a macro. type the statements, one line at a time. when done, type 'end'
> greet $name
> reply xyz
> end
>> macro "introduce $name" has been added
$

```

Reply:
ok hasitha il work on it

Reply:
Hi hasitha, just to let you know i implemented it on the command line in kasaya in the prototype branch using the above function, but i'll implement it on jarvis as well.